src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -322,6 +322,11 @@ public function ItemEditForm()
         if ($cb) {
             $cb($form, $this);
         }
+
+        if (is_a($this->record, DataObject::class) && $this->record->getRequireSudoMode()) {
```

Comment:
Scenario is the edit field in a gridfield

Reply:
This should be after `updateItemEditForm` because that extension hook might add additional fields, which we want to be included in the readonly transformation.

Reply:
Updated